src/hotspot/share/gc/g1/g1HeapRegionAttr.hpp

```diff
@@ -131,9 +131,11 @@ class G1HeapRegionAttrBiasedMappedArray : public G1BiasedMappedArray<G1HeapRegio
     get_ref_by_index(index)->set_new_survivor();
   }
 
-  void set_humongous_candidate(uintptr_t index, bool remset_is_tracked) {
+  void set_humongous_candidate(uintptr_t index) {
     assert(get_by_index(index).is_default(),
            "Region attributes at index " INTPTR_FORMAT " should be default but is %s", index, get_by_index(index).get_type_str());
+    // Humongous candidates must have complete remset.
+    const bool remset_is_tracked = true;
```

Comment:
Wouldn't it be preferable directly asserting this instead of only "requiring" this in a comment?

Reply:
That would be ideal, but I don't see an easy way retrieving `HeapRegion` by index in this context.